sklearn/svm/tests/test_sparse.py

```diff
@@ -161,46 +162,49 @@ def scramble_indices(X):
     )
 
 
-def test_svc_with_custom_kernel():
+@pytest.mark.parametrize("lil_container", LIL_CONTAINERS)
+def test_svc_with_custom_kernel(lil_container):
     def kfunc(x, y):
         return safe_sparse_dot(x, y.T)
 
+    X_sp = lil_container(X)
     clf_lin = svm.SVC(kernel="linear").fit(X_sp, Y)
     clf_mylin = svm.SVC(kernel=kfunc).fit(X_sp, Y)
     assert_array_equal(clf_lin.predict(X_sp), clf_mylin.predict(X_sp))
 
 
 @skip_if_32bit
-def test_svc_iris():
+@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
+@pytest.mark.parametrize("kernel", ["linear", "poly", "rbf"])
+def test_svc_iris(csr_container, kernel):
     # Test the sparse SVC with the iris dataset
-    for k in ("linear", "poly", "rbf"):
-        sp_clf = svm.SVC(kernel=k).fit(iris.data, iris.target)
-        clf = svm.SVC(kernel=k).fit(iris.data.toarray(), iris.target)
+    iris_data_sp = csr_container(iris.data)
 
-        assert_array_almost_equal(
-            clf.support_vectors_, sp_clf.support_vectors_.toarray()
-        )
-        assert_array_almost_equal(clf.dual_coef_, sp_clf.dual_coef_.toarray())
-        assert_array_almost_equal(
-            clf.predict(iris.data.toarray()), sp_clf.predict(iris.data)
-        )
-        if k == "linear":
-            assert_array_almost_equal(clf.coef_, sp_clf.coef_.toarray())
+    sp_clf = svm.SVC(kernel=kernel).fit(iris_data_sp, iris.target)
+    clf = svm.SVC(kernel=kernel).fit(iris.data, iris.target)
+
+    assert_array_almost_equal(clf.support_vectors_, sp_clf.support_vectors_.toarray())
```

Comment:
Could you replace all instances of `assert_array_almost_equal` by `assert_allclose`?